x/concentrated-liquidity/lp.go

```diff
@@ -307,6 +305,11 @@ func (k Keeper) WithdrawPosition(ctx sdk.Context, owner sdk.AccAddress, position
 			return osmomath.Int{}, osmomath.Int{}, err
 		}
 
+		// Note that here we currently use the iterator based definition to search
+		// for a remaining position in the pool. Since we have removed a position we need to
+		// search if there are more.
+		// Ideally in the future we could have a simple "num positions" counter to make this logic
```

Comment:
Do you mean adding a state entry for num positions where key is address and value is num positions?

Reply:
Yes! But key is poolID

Reply:
ah right pool id. Would the speedup of using num counter be worthy for making an github issue and coming back to it to address it?

Reply:
I think so, but there are other higher priorities in the CL performance line of work.